apps/researcher/src/lib/route-segment-transformer.test.ts

```diff
@@ -0,0 +1,21 @@
+import {describe, expect, it} from '@jest/globals';
+import {
+  encodeRouteSegment,
+  decodeRouteSegment,
+} from './route-segment-transformer';
+
+describe('encodeRouteSegment', () => {
```

Comment:
For consideration: the heart of `route-segment-transformer.ts` is about replacing dots. Should we have a test for this, or rename the current test?